drivers/spi/spi_sifive.h

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2018 SiFive Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _SPI_SIFIVE__H
+#define _SPI_SIFIVE__H
+
+#define SET_BIT(reg, offset)    ((reg) |= (1 << (offset)))
+#define CLR_BIT(reg, offset)    ((reg) &= ~(1 << (offset)))
+#define CLR_MASK(reg, mask)     ((reg) &= ~(mask))
+
+#define SPI_CFG(dev) ((struct spi_sifive_cfg *) ((dev)->config->config_info))
+#define SPI_DATA(dev) ((struct spi_sifive_data *) ((dev)->driver_data))
+
+#define _REG32(p, i) (*(volatile u32_t *) ((p) + (i)))
+#define SPI_REG(dev, offset) _REG32(SPI_CFG(dev)->base, offset)
```

Comment:
seems like your are rewriting something that could be done with sys_io.h function. Change this to sys_<write/read>32(). (reason being if these is a better way - it's most probably very unlikely here though -  to access the memory, one change in sys_io implementation would benefit everywhere, and you would not have to do any change here)